### PR TITLE
RHCLOUD-35642: Port setting additional environment variables for IQE

### PR DESCRIPTION
**Pull request by skarekrow**
This is needed to allow running in Konflux with additional environment variables.

Logic is ported away from using bash to something more portable with awk. In addition this should be easier to maintain because the purpose is more clear and the logic is easier to follow.

Comment included from original script so others know this is a comma delimited string.

### CODE REVIEW

File: konflux_scripts/deploy-iqe-cji.sh [hunk 1]
```diff
@@ -26,13 +26,22 @@ main() {
...
         selenium_arg="--selenium"
     fi
 
+    iqe_env_var_args=$(awk -v IQE_ENV_VARS="$iqe_env_vars" 'BEGIN {
+      split(IQE_ENV_VARS, iqe_env_vars, ",");
+      for (i in iqe_env_vars) {
+        printf "--env-var " iqe_env_vars[i] " "
+      }
+    }')
+
```
Comment (Victoremepunto) [inline konflux_scripts/deploy-iqe-cji.sh:44]: this is nice, thanks for bringing back support for env vars.

The only think I'd be suggesting would be to avoid the usage of `awk` in favor of bash arrays. 

check this for reference of what I mean. Basically is to iterate over a list of items, using the different elements of an array: 

https://github.com/RedHatInsights/cicd-tools/blob/d9c2a58b29caa16afa250faed7a0d44e987b11a8/src/shared/image_builder.sh#L61-L63

Comment (skarekrow) [reply konflux_scripts/deploy-iqe-cji.sh:44]: I'm personally in favor of doing arrays in awk as the splitting logic is clearer and we don't need to mess with the shell's IFS to do so. It's already available on the system and in case we drop support for bash due to some reason in the future (moving to POSIX shell, busybox or dash) to cut down on surface area of attacks, I think it makes sense this isn't written bash specific.

Comment (Victoremepunto) [reply konflux_scripts/deploy-iqe-cji.sh:44]: Not wanting to use IFS, I was trying to make the point of using a Bash array, provided the value of the variable is a spaced separated list of values. That way you could have something like 

```
IQE_ENV_VARS=("FOO=foo" "BAR=bar")

for env_var in "${IQE_ENV_VARS[@]}"; do
  env_var_params+=('--env-var' "$env_var");
done

# output is 
--env-var FOO=foo --env-var BAR=bar
```

if the comma separated values are mandatory, then I guess that option is okay ? I rarely see `awk` as an option for simply iterating values but I do agree is more readable and preferrable than to use IFS

Comment (skarekrow) [reply konflux_scripts/deploy-iqe-cji.sh:44]: If we're willing to break backwards behavior, I would agree with you. But since the requirement currently is comma delimited values (which will require changing the IFS to use bash arrays), I found this cleaner than the bash solution. I feel awk is the right tool for the job here, if you disagree and we want to make a backwards breaking change I'll move this over to using bash instead.

TLDR; `awk` is already there and great for this, let's use it :)

File: konflux_scripts/deploy-iqe-cji.sh [hunk 2]
```diff
@@ -26,13 +26,22 @@ main() {
     local iqe_test_importance="${IQE_TEST_IMPORTANCE}"
     local iqe_plugins="${IQE_PLUGINS}"
     local iqe_env="${IQE_ENV:-clowder_smoke}"
+    #iqe_env_vars="ENV_VAR1=value1,ENV_VAR2=value2" -- custom set of extra environment variables to set on IQE pod
+    local iqe_env_vars="${IQE_ENV_VARS}"
```
Comment (Victoremepunto) [inline konflux_scripts/deploy-iqe-cji.sh:30]: why is this initialized to `IQE_ENV_VARS` ?

Comment (Victoremepunto) [reply konflux_scripts/deploy-iqe-cji.sh:30]: Oh well, I guess, readability. that's fair

File: konflux_scripts/deploy-iqe-cji.sh (hunk 1 above)
Comment (Victoremepunto) [reply konflux_scripts/deploy-iqe-cji.sh:44]: since we have to keep the `,` as a separator, I believe its a fair use :) , prefer this to use the IFS

**Review by Victoremepunto — APPROVED** [review]

**Comment by bsquizz** [comment]
The old pr check scripts support this variable. The konflux ones are out of sync

**Comment by bsquizz** [comment]
See https://github.com/RedHatInsights/cicd-tools/blob/main/cji_smoke_test.sh#L18
